fix: refine query ETA display and Kalman filter stability

rustyconover opened the pull request:
## UI

- Hide ETA when the Kalman filter has no valid prediction.
- Clamp ETA to 0.5s when query progress > 99.9% to reduce oscillations.

## Kalman Filter
- When progress bar delay = 0, defer ETA predictions until enough data is collected to estimate velocity reliably.
- Track last query progress percentage and skip duplicate values (prevents filter instability).
- Tune Alpha, P, Q, and R via differential evolution using sample queries. Improves accuracy and avoids numerical instability.

rustyconover committed:
fix: refine query ETA display and Kalman filter stability

- Hide ETA when the Kalman filter has no valid prediction.
- Clamp ETA to 0.5s when query progress > 99.9% to reduce oscillations.

- When progress bar delay = 0, defer ETA predictions until enough data is collected to estimate velocity reliably.
- Track last query progress percentage and skip duplicate values (prevents filter instability).
- Tune Alpha, P, Q, and R via differential evolution using sample queries. Improves accuracy and avoids numerical instability.

rustyconover commented:
@Mytherin here is an example of the improvements on a chart for this query:

```sql
SUMMARIZE FROM read_parquet(repeat(['/Users/rusty/Development/discreture/example.parquet'], 3));
```

<img width="1188" height="490" alt="image" src="https://github.com/user-attachments/assets/0785437e-b7f5-44a2-a4bd-ff5cf0640a2f" />

You'll see the updated filter parameters result in a much smoother projected ETA compared to the original values.

rustyconover commented:
@carlopi please give this a spin, I'm interested in what you think too!

Mytherin reviewed (commented):
Thanks! This seems to work better, but there longer queries still seem to not have an estimate. For example:

```sql
SUMMARIZE FROM read_parquet(repeat(['~/Data/hits.parquet'], 100));
```

I wonder if the constants that you are adjusting here could be set dynamically based on the initial estimate instead?

rustyconover committed:
fix: improve Kalman filter stability and scaling

* Scale progress inputs relative to initial P, Q, R matrix parameters to fix
  irrational results from small progress updates
* Add dynamic scaling factor to handle both short and long duration queries
* Update ALPHA, P, Q, R settings based on zero-delay test fitting
* Skip Kalman filter initialization when initial progress is zero

Fixes issue where Kalman filter sensitivity to input scale caused unstable
behavior with small progress values.

rustyconover commented on src/common/progress_bar/terminal_progress_bar_display.cpp:
```diff
@@ -120,9 +118,33 @@ void TerminalProgressBarDisplay::Update(double percentage) {
+	//
+	// For the interval where there are no estimates, no ETA will be shown, but
+	// the progress bar will appear.
+	if (current_time > 0.5 && filter_percentage > 0.01) {
```
I've removed delay now.  I've changed `Initialize` on the Kalman filter to just skip initialization if the velocity is zero.  Now even long running queries get an estimated ETA quickly.

Mytherin commented:
Thanks!
